packages/web/src/app/doc-pages/components/components-info.ts

```diff
@@ -0,0 +1,96 @@
+import { DocPageName, Pages } from 'src/app/shared/shared.enum';
+
+interface ComponentInfo {
+  title?: string;
+  components: DocPageName[];
+}
+
+interface ComponentInfoList {
+  [category: string]: ComponentInfo[];
+}
+
```

Comment:
Would this work?

```js
export type Category = 'navigation' | 'action' | 'input' | 'dataDisplay' | 'feedback' | 'layout';

type ComponentInfoList = Record<Category, ComponentInfo[]>;
```

Reply:
This should also work, depending on what syntax you like the more (`Record<,>` vs `interface`):
```suggestion
export type Category = 'navigation' | 'action' | 'input' | 'dataDisplay' | 'feedback' | 'layout';

interface ComponentInfoList {
  [category: Category]: ComponentInfo[];
}

```

Reply:
Also, since the `ComponentInfoList` isn't actually a list, but rather an object, consider renaming it to `ComponentInfo` or something like that